apps/designer/app/designer/features/props-panel/props-panel.tsx

```diff
@@ -10,19 +10,99 @@ import type { SelectedInstanceData } from "~/shared/canvas-components";
 import {
   Box,
   Button,
-  Combobox,
-  ComboboxTextField,
-  ComboboxPopperContent,
   Grid,
+  TextField,
   Tooltip,
+  useCombobox,
+  ComboboxPrimitive,
+  ComboboxContent,
+  ComboboxAnchor,
+  ComboboxListbox,
+  ComboboxListboxItem,
+  IconButton,
 } from "@webstudio-is/design-system";
 import {
   PlusIcon,
   TrashIcon,
   ExclamationTriangleIcon,
+  ChevronDownIcon,
 } from "@webstudio-is/icons";
 import { handleChangePropType, usePropsLogic } from "./use-props-logic";
 
+type ComboboxProps = {
+  isReadonly: boolean;
+  isInvalid: boolean;
+  items: Array<string>;
+  value: string;
+  onItemSelect: (value: string | null) => void;
+  onSubmit: (value: string) => void;
+};
+
+export const Combobox = ({
+  isReadonly,
+  isInvalid,
+  items: itemsProp,
+  value,
+  onItemSelect,
+  onSubmit,
+}: ComboboxProps) => {
+  const {
+    items,
+    getInputProps,
+    getComboboxProps,
+    getToggleButtonProps,
+    getMenuProps,
+    getItemProps,
+    isOpen,
+  } = useCombobox({
+    items: itemsProp,
+    value,
+    itemToString: (item) => item ?? "",
+    onItemSelect,
+  });
+
+  return (
+    <ComboboxPrimitive>
```

Comment:
I don’t like this API. They hide the real thing with no benefit. What is primitive and what it’s doing? What is anchor? Keep it as is by using Popper?

Reply:
I could call it ComboboxPopper etc to make it more clear, I think it makes sense to reexport just to make sure we can do something in combobox later if needed

Reply:
changed